x/delegation/keeper/un_delegation_state.go

```
@@ -141,3 +150,29 @@ func (k Keeper) GetWaitCompleteUndelegationRecords(ctx sdk.Context, height uint6
 	// The states of records stored by WaitCompleteUndelegations kvStore should always be IsPending,so using AllRecords as getType here is ok.
 	return k.GetUndelegationRecords(ctx, recordKeys, AllRecords)
 }
+
+func (k *Keeper) IncrementUndelegationHoldCount(ctx sdk.Context, recordKey []byte) {
+	prev := k.GetUndelegationHoldCount(ctx, recordKey)
+	if prev == math.MaxUint64 {
+		panic("cannot increment undelegation hold count above max uint64")
```

It is better to remove the `panic`s happening here and instead return an error.

---

Yeah, the hook interface might also need to return an error.